File: 2021/7.py
from Utils import *

# Part 1

inp = rd("7.txt")

# inp = "16,1,2,0,4,2,7,1,2,14"

inp = ints(inp)

c = 0
min_c = math.inf
for i in range(450, 600):
    c = 0
    for x in inp:
        f = 1
        while x != i:
            if x > i:
                x -= 1
            else:
                x += 1
            c += f
            f += 1

    if c < min_c:
        min_c = c
        print(min_c)

print(min_c)

File: 2021/7.R
source("Utils.R")

inp <- as.numeric(strsplit(readLines("2021/7.txt"), ",")[[1]])

min(sapply(min(inp):max(inp), function(x) sum(abs(inp - x))))
min(sapply(min(inp):max(inp), function(x) sum(sapply(inp, function(y) sum(abs(y-x):1)))))

